src/OpenTK.Platform/Native/Windows/WindowComponent.cs

```diff
@@ -1143,21 +1107,88 @@ private IntPtr Win32WindowProc(IntPtr hWnd, WM uMsg, UIntPtr wParam, IntPtr lPar
             }
         }
 
+        /// <summary>
+        /// Processes raw input events using GetRawInputBuffer.
+        /// </summary>
+        private void ProcessRawInputBuffer()
+        {
+            unsafe
+            {
+                var rawInputMessagesProcessed = 0;
+                uint rawInputHeaderSize = (uint)Marshal.SizeOf<Win32.RAWINPUTHEADER>();
+                uint rawInputBufferSize = 0;
+                uint queryCountResult = Win32.GetRawInputBuffer(null, ref rawInputBufferSize, rawInputHeaderSize);
+                if (queryCountResult == unchecked((uint)-1))
+                {
+                    throw new Win32Exception(Marshal.GetLastWin32Error(), "GetRawInputBuffer (query size) failed.");
+                }
+
+                //Slightly oversize the buffer to reduce chance of ERROR_INSUFFICIENT_BUFFER occurring
+                //GLFW uses this technique
+                rawInputBufferSize *= 2;
+                if (rawInputBufferSize > _rawInputBufferSize || _rawInputBuffer == null)
+                {
+                    if (_rawInputBuffer == null)
+                    {
+                        _rawInputBuffer = (Win32.RAWINPUT*)NativeMemory.Alloc(rawInputBufferSize);
+                    }
+                    else
+                    {
+                        _rawInputBuffer = (Win32.RAWINPUT*)NativeMemory.Realloc(_rawInputBuffer, rawInputBufferSize);
+                    }
+                    _rawInputBufferSize = rawInputBufferSize;
+                }
+
+                rawInputBufferSize = _rawInputBufferSize;
+                uint elementsRead = Win32.GetRawInputBuffer(_rawInputBuffer, ref rawInputBufferSize, rawInputHeaderSize);
+                if (elementsRead == unchecked((uint)-1))
+                {
+                    int error = Marshal.GetLastWin32Error();
+                    if (error != Win32.ERROR_INSUFFICIENT_BUFFER)
+                    {
+                        throw new Win32Exception(error, "GetRawInputBuffer (fetch data) failed.");
+                    }
```

Comment:
This should not be an error, we should handle this properly. If you want I can rewrite this if you don't want to.